apps/api/src/app/notification-template/notification-template.controller.ts

```diff
@@ -56,12 +59,17 @@ export class NotificationTemplateController {
     summary: 'Get notification templates',
   })
   @ExternalApiAccessible()
-  getNotificationTemplates(@UserSession() user: IJwtPayload): Promise<NotificationTemplateResponse[]> {
+  getNotificationTemplates(
+    @UserSession() user: IJwtPayload,
+    @Query() query: NotificationTemplatesRequestDto
+  ): Promise<NotificationTemplatesResponseDto> {
     return this.getNotificationTemplatesUsecase.execute(
       GetNotificationTemplatesCommand.create({
         organizationId: user.organizationId,
         userId: user._id,
         environmentId: user.environmentId,
+        page: query.page ? Number(query.page) : 0,
+        usePagination: query.usePagination === 0 ? false : true,
```

Comment:
I think that if we are going this direction we can just enable pagination by default, and instead of passing usePagination, we can allow the user to pass the `limit` property which by default would be for example 15 entries~ and maximum value allowed would be `100` for some API usecases. Wdyt?